velox/dwio/common/ColumnVisitors.h

```diff
@@ -93,18 +93,21 @@ class ExtractToHook {
   template <typename T>
   void addNull(vector_size_t rowIndex) {
     hook_.addNull(rowIndex);
+    reader_->incrementNumValues(1);
```

Comment:
numValues is recording the values inside result vector, in this case it is 0.  Should the hook itself keep track of the number of values instead?

Reply:
Hi Yuhta, thanks for reviewing.

> numValues is recording the values inside result vector, in this case it is 0.

I was trying to understand this. Would you mean we don't have to do the increment here?

> Should the hook itself keep track of the number of values instead?

I expected a solution like this but it will involve a broader scale of changes in the code base. So far it's safe enough to advance the reader directly like the PR does, as the raw values in the reader doesn't really matter if hook is specified. `numValues_` will only work like a counter in the case.

Reply:
Can you point to me where are we depending on the reader numValues for hook?  My impression is hook does not rely on it at all?

Reply:
@Yuhta 
 
https://github.com/facebookincubator/velox/blob/0c3838716786ecad0950161ec14190f49f4d458a/velox/dwio/parquet/reader/PageReader.h#L549-L554

It's line 551 in `PageReader.h`. Because the `numValuesBeforePage` is wrong, the `numValuesBias` of the visitor will be wrong. Then the offset of the next value being sent to hook will be calculated wrong:

https://github.com/facebookincubator/velox/blob/42193a8015081187e06ed4e8ed77b2bb1002a236/velox/dwio/common/ColumnVisitors.h#L522-L532